lib/protocol/http/header/cache_control.rb

```diff
@@ -55,11 +59,27 @@ def no_store?
 					self.include?(NO_STORE)
 				end
 				
+				def must_revalidate?
+					self.include?(MUST_REVALIDATE)
+				end
+				
+				def proxy_revalidate?
+					self.include?(PROXY_REVALIDATE)
+				end
+				
 				def max_age
 					if value = self.find{|value| value.start_with?(MAX_AGE)}
 						_, age = value.split('=', 2)
 						
-						return Integer(age)
+						return Integer(age, exception: false)
+					end
+				end
+				
+				def s_maxage
```

Comment:
On the one hand, this is the exact name from the spec. On the other hand, its ugly and non-obvious.

Technically, it would make more sense `shared_max_age` or `shared_maximum_age`. Alternatively, we should start copying in documentation.

I don't have a strong opinion, just that initially it looks inconsistent `maxage` vs `max-age`, non-obvious.

Reply:
I interpreted CacheControl as presently following a nomenclature of translating attributes with `-` to methods with `_`, but otherwise keeping names the same, so I was aspiring to keep with that. I could have interpreted that wrongly, of course.

I'm happy to change if you want.

I'm not sure what you mean by "copying in documentation". Were you thinking adding RFC references, or did you have something else in mind?

Reply:
Yeah, RFC, or Mozilla docs, etc. Like, I didn't know what `s_maxage` was and I had to go look up the docs.

> s-maxage
> The s-maxage response directive also indicates how long the response is [fresh](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#fresh_and_stale_based_on_age) for (similar to max-age) — but it is specific to shared caches, and they will ignore max-age when it is present.

(from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)

Reply:
Well, there is value in calling it what it is, even if it is "consistently unclear" at best. OTOH, we can call it whatever we want. I just don't have a strong opinion. Maybe introducing our own snowflake name is a bad idea too.

Reply:
I've added brief comments plus RFC links. Left the names as-is.